raiden/tests/utils/transfer.py

```diff
@@ -115,26 +115,27 @@ def get_channelstate(
 
 
 @contextmanager
-def watch_for_unlock_failures(*apps, retry_timeout=DEFAULT_RETRY_TIMEOUT):
+def watch_for_unlock_failures(*apps):
     """
     Context manager to assure there are no failing unlocks during transfers in integration tests.
     """
 
-    def watcher_function():
-        offset = {
-            app.raiden.address: count_unlock_failures(app.raiden.wal.storage.get_events())
-            for app in apps
-        }
-        while True:
-            for app in apps:
-                assert not has_unlock_failure(app.raiden, offset=offset[app.raiden.address])
-            gevent.sleep(retry_timeout)
+    failed_event = None
```

Comment:
I would like to just assert inside `check()` here, but then the `AssertionError` happens inside the `RaidenService._handle_event` subtask and does not make the test fail. It seems to be a general problem (errors in a subtask only crashing one node in the `raiden_network` fixture instead of the whole test), for which I will open a follow-up issue.

Reply:
This should not be correct. This decorator has to be used on the integration tests to fix this problem:

https://github.com/raiden-network/raiden/blob/30b924f2399600827251dbc16fdf90cbb3dcb8e3/raiden/tests/utils/detect_failure.py#L14-L19

And if any of the underlying nodes dies the test will fail.